contributors/design-proposals/federated-statefulsets.md

```diff
@@ -0,0 +1,248 @@
+# Federated StatefulSet design
+irfan.rehman@huawei.com, dhilip.kumar.s@huawei.com
+
+# Goal
+
+To outline the use cases for the need of federated Statefulset and determine suitable design/implementation approach for the same.
+
+# Background
+
+The original design proposal of the statefulset in kubernetes, including the use cases and example applications, can be found [here](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/stateful-apps.md).
+
+
+# Use Cases
+
+1 – A stateful app, for the reasons of high availability, wants the stateful pods distributed in different clusters, such that the set can withstand cluster failures. This represents an app with one single global quorum.
+
+2 – A stateful app, wants replicas distributed in multiple clusters, such that it can form multiple smaller sized local clusters using only local replicas.
+
+# Design Requirements
+
+If we consider the use cases listed above, the main design requirements can roughly be listed as:
+
+1 – An unique, consistent and discoverable identity for each replica/instance across the federated clusters.
```

Comment:
I'd like a new design requirement.

2. It must be possible for the federated sets to SAFELY form an initial quorum that adds the rest of the set.  

Any design which doesn't allow (for instance) cluster 1 to form an initial quorum and then safely add members in cluster 2 (or any other order, really), is a non-starter because then stateful set at federation has a different set of guarantees.

I think that's implicit in some of your comments, but it needs to be explicit, obvious, and impossible to break.